### ISSUE TITLE
Handle variable length trials and data with NaN values

### ISSUE THREAD

**Issue by ahwillia** [ganguli-lab/twpca#5]
Trial lengths can be variable in some experiments. We should extend twPCA to handle this by simply ignoring `NaN` values in the data tensor. Users still specify the data as a `trials x time x neuron` array, but indicate missing data with nans.

Another possibility would be to pass in a vector holding the indices of trial start or end:

```python
model = TWPCA(n_components).fit(data, trial_start=..., trial_end=...)
```

**Comment by poolio** [comment]
Things that need to get fixed:
- Low-rank factor initialization. Need to replace SVD with matrix completion or other strategy.
- Initial time warps (linear scaling to same length? still use shifts?)
- Mask loss function to only consider valid timepoints

**Comment by poolio** [comment]
Fixed in #6
